### PR TITLE
Accordions

### PR DESCRIPTION
**Pull request by illright**

### CODE REVIEW

File: src/accordion/accordion-section.svelte [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  import classes from '../_utils/classes.js';
+
+  let _class = null;
+  export { _class as class };
+
+  export let label = null;
+  export let open = false;
+
+  const selfControl = {
+    close() {
```
Comment (aabounegm) [inline src/accordion/accordion-section.svelte:13]: So there is `close` and `toggle` but not `open`?

File: src/accordion/accordion-section.svelte [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      open = false;
+    },
+    toggle() {
+      if (!open) {
+        dispatch('panel-open', selfControl);
```
Comment (aabounegm) [inline src/accordion/accordion-section.svelte:18]: Why not dispatching when it is closed as well? Also, where is the native event?

File: src/accordion/accordion-section.svelte [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  import { createEventDispatcher } from 'svelte';
+  import classes from '../_utils/classes.js';
+
+  let _class = null;
+  export { _class as class };
+
+  export let label = null;
+  export let open = false;
+
+  const selfControl = {
```
Comment (aabounegm) [inline src/accordion/accordion-section.svelte:12]: I don't really see the need for an object here. It's not like we have a lot of them and need to organize internal states separately

File: src/accordion/accordion-section.svelte (hunk 1 above)
Comment (illright) [reply src/accordion/accordion-section.svelte:13]: This `selfControl` is a piece of internal API anyway, and only `toggle` is exposed as a slot prop, so no problem here. `open` is not used internally.

File: src/accordion/accordion-section.svelte (hunk 3 above)
Comment (illright) [reply src/accordion/accordion-section.svelte:12]: The need for an object is simply for guaranteed identification of each panel. Objects have referential equality, therefore, each panel is uniquely identified by its `selfControl` object, which is the behaviour that `<Accordion>` relies on

File: src/accordion/accordion-section.svelte (hunk 2 above)
Comment (illright) [reply src/accordion/accordion-section.svelte]: Technically, this is an internal event that wasn't really meant to be listened on, but now that I think about it, why not. Where's the native event, you ask? There might not be. We don't know what has triggered the toggle of the panel, it's just a callback that a user calls at their liking

File: demo/App.svelte [hunk 4]
```diff
@@ -36,9 +38,9 @@
   <Button filled>
     Button!!
   </Button>
-  <RadioGroup items={colorItems} color bind:value={colorItem} name="colors" />
+  <!-- <RadioGroup items={colorItems} color bind:value={colorItem} name="colors" />
   <RadioGroup items={items} bind:value={item} name="numbers" />
-  <CheckboxGroup items={items} max={1} name="numbers-check" />
+  <CheckboxGroup items={items} max={1} name="numbers-check" /> -->
```
Comment (aabounegm) [inline demo/App.svelte:43]: Why is this commented out?

Comment (illright) [reply demo/App.svelte:43]: It was too long on the page and making me sick. This isn't even a proper demo page anyway, I just needed some space to showcase the components from the latest PRs. It's okay to leave this commented.

**Pull request merged**